File: Lesson2/4_A/CSharp/CSharp/Form1.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Windows.Forms;

namespace CSharp
{
    public partial class Form1 : Form
    {
        //params:
        static int max = 120;
        static int min = 0;



        //---------------------------
        Random R = new Random();
        double avg = 0;
        int count = 0;
        int[] freq = new int[max];

        public Form1()
        {
            InitializeComponent();
            start();
        }

        private void start()
        {
            for(int i=0; i<max; i++)
            {
                freq[i] = 0;
            }
        }

        private void timer1_Tick(object sender, EventArgs e)
        {
            count++;
            int random = R.Next(min, max);
            avg = avg + (random - avg) / count;
            freq[random]++;
            richTextBox1.AppendText( "age: " + random + "  new average: " + avg + Environment.NewLine);
        }

        private void button1_Click(object sender, EventArgs e)
        {
            timer1.Start();
            richTextBox1.AppendText("---------------------------------------------" + Environment.NewLine);
        }

        private void button2_Click(object sender, EventArgs e)
        {
            timer1.Stop();
            richTextBox1.AppendText("---------------------------------------------" + Environment.NewLine);
            for (int i = 0; i < max; i++)
            {
                richTextBox1.AppendText("Age: " + i + "  count: " + freq[i] + Environment.NewLine);
            }
        }
    }
}

File: Lesson2/4_A/VB_Net/VB_Net/Form1.vb
Public Class Form1

    'params'
    Dim max As Integer = 120
    Dim min As Integer = 0
    '----------------------------------------------'

    Dim R As New Random
    Dim avg As Double = 0
    Dim list As New List(Of Integer)
    Dim done As Boolean = False
    Dim n As Integer = 0

    Private Sub Timer1_Tick(sender As Object, e As EventArgs) Handles Timer1.Tick
        n += 1
        Dim random As Integer = R.Next(min, max)
        avg = avg + (random - avg) / n
        list(random) += 1
        Me.RichTextBox1.AppendText("Age: " & random & "  Average: " & avg & Environment.NewLine)

    End Sub

    Private Sub Button1_Click(sender As Object, e As EventArgs) Handles Button1.Click
        If done = False Then
            For i As Integer = 0 To (max - 1)
                list.Insert(i, 0)
            Next
            done = True
        End If

        Me.RichTextBox1.AppendText("------------------------------------------------" & Environment.NewLine)
        Me.Timer1.Start()

    End Sub

    Private Sub Button2_Click(sender As Object, e As EventArgs) Handles Button2.Click
        Me.Timer1.Stop()
        Me.RichTextBox1.AppendText("------------------------------------------------" & Environment.NewLine)
        For i As Integer = 0 To (max - 1)
            Me.RichTextBox1.AppendText("Age: " & i & "  Count: " & list(i) & Environment.NewLine)
        Next

    End Sub
End Class
